core/src/test/java/com/sequenceiq/cloudbreak/service/image/ImageCatalogServiceDefaultNotFoundTest.java

```diff
@@ -61,10 +64,29 @@
     @Mock
     private LatestDefaultImageUuidProvider latestDefaultImageUuidProvider;
 
+    @InjectMocks
+    private VersionBasedImageProvider versionBasedImageProvider;
+
+    @Mock
+    private AdvertisedImageProvider advertisedImageProvider;
+
+    @InjectMocks
+    private ImageCatalogServiceProxy imageCatalogServiceProxy;
+
+    @InjectMocks
+    private AdvertisedImageCatalogService advertisedImageCatalogService;
+
+    @InjectMocks
+    private VersionBasedImageCatalogService versionBasedImageCatalogService;
+
     @Before
     public void beforeTest() {
         ReflectionTestUtils.setField(underTest, "cbVersion", "5.0.0");
         ReflectionTestUtils.setField(underTest, "defaultCatalogUrl", "");
+        ReflectionTestUtils.setField(underTest, "imageCatalogServiceProxy", imageCatalogServiceProxy);
+        ReflectionTestUtils.setField(imageCatalogServiceProxy, "advertisedImageCatalogService", advertisedImageCatalogService);
```

Comment:
@InjectMocks and @Mock annotations should take care of these

Reply:
It is true, when you would inject mocks but in this case I'm "injecting" real implementations with mocked dependencies. This way we can make sure the refactored implementation is in line with the original one.